### PR TITLE
Mission Autotest - Add AP Units, Group size and Vehicle checks

### PR DESCRIPTION
**Pull request by Mike-MF**
Surprisingly it works.

- For AP it checks for `TACU_isAP = 1` (6 is warning, 10 is error)
- Group count checks for groups with more than 10 AI in them.
- Wheeled APCs (4 is warning, 6 is error)
- Tracked APCs & Tanks are considered the same, will use the same counter. (2 is warning, 4 is error)

Also, my translations are incredible.

Tested all parts, they work just fine.

Untitled

### CODE REVIEW

**Comment by jonpas** [comment]
If you're not sure about translations, best to remove them.

**Comment by Mike-MF** [comment]
The translations should actually be pretty accurate.

File: addons/mission_autotest/functions/fnc_testAPAmount.sqf [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#include "script_component.hpp"
+/*
+ * Author: TMF Team, Jonpas, Mike
+ * Tests AP AI amount
```
Comment (jonpas) [inline addons/mission_autotest/functions/fnc_testAPAmount.sqf:4]:
```suggestion
 * Tests AP AI amount.
```

File: addons/mission_autotest/functions/fnc_testAPAmount.sqf [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#include "script_component.hpp"
+/*
+ * Author: TMF Team, Jonpas, Mike
```
Comment (jonpas) [inline addons/mission_autotest/functions/fnc_testAPAmount.sqf:3]:
```suggestion
 * Author: Mike
```

File: addons/mission_autotest/functions/fnc_testAPAmount.sqf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [] call tac_mission_autotest_fnc_testAPAmount
+ *
+ * Public: No
+ */
+
+private _apAmount = count (allUnits select {(getNumber (configFile >> "CfgVehicles" >> typeOf _x >> "TACU_isAP")) == 1});
```
Comment (jonpas) [inline addons/mission_autotest/functions/fnc_testAPAmount.sqf:18]:
```suggestion
private _apAmount = count (allUnits select {
    getNumber (configFile >> "CfgVehicles" >> typeOf _x >> "TACU_isAP") == 1
});
```

File: addons/mission_autotest/functions/fnc_testAPCAmount.sqf [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#include "script_component.hpp"
+/*
+ * Author: TMF Team, Jonpas, Mike
```
Comment (jonpas) [inline addons/mission_autotest/functions/fnc_testAPCAmount.sqf:3]:
```suggestion
 * Author: Mike
```

File: addons/mission_autotest/functions/fnc_testGroupCount.sqf [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#include "script_component.hpp"
+/*
+ * Author: TMF Team, Jonpas, Mike
```
Comment (jonpas) [inline addons/mission_autotest/functions/fnc_testGroupCount.sqf:3]:
```suggestion
 * Author: Mike
```

File: addons/mission_autotest/functions/fnc_testTankAmount.sqf [hunk 6]
```diff
@@ -0,0 +1,31 @@
+#include "script_component.hpp"
+/*
+ * Author: TMF Team, Jonpas, Mike
```
Comment (jonpas) [inline addons/mission_autotest/functions/fnc_testTankAmount.sqf:3]:
```suggestion
 * Author: Mike
```

File: addons/mission_autotest/stringtable.xml [hunk 7]
```diff
@@ -103,6 +103,46 @@
...
+            <German>Prüft, ob die empfohlene Anzahl von AP-Einheiten vorhanden ist.</German>
+            <French>Vérifie si la quantité recommandée d'unités AP est présente.</French>
+        </Key>
+        <Key ID="STR_TAC_Mission_Autotest_APCAmount">
+            <English>Wheeled APCs:</English>
+            <German>APC mit Rädern:</German>
+            <French>Véhicules de transport de troupes à roues:</French>
+        </Key>
+        <Key ID="STR_TAC_Mission_Autotest_APCAmountDesc">
+            <English>Checks if the recommended amount of Wheeled APC vehicles are present.</English>
```
Comment (jonpas) [inline addons/mission_autotest/stringtable.xml:122]:
```suggestion
            <English>Checks if the recommended amount of Wheeled APCs are present.</English>
```

File: addons/mission_autotest/stringtable.xml [hunk 8]
```diff
@@ -103,6 +103,46 @@
...
+            <German>Überprüft, ob die empfohlene Anzahl an Wheeled APC Fahrzeugen vorhanden ist.</German>
+            <French>Vérifie si le nombre recommandé de véhicules APC à roues est présent.</French>
+        </Key>
+        <Key ID="STR_TAC_Mission_Autotest_TankAmount">
+            <English>Tracked Vehicles:</English>
+            <German>Raupenfahrzeuge:</German>
+            <French>Véhicules à chenilles</French>
+        </Key>
+        <Key ID="STR_TAC_Mission_Autotest_TankAmountDesc">
+            <English>Checks if the recommended amount of Tracked vehicles are present.</English>
```
Comment (jonpas) [inline addons/mission_autotest/stringtable.xml:132]:
```suggestion
            <English>Checks if the recommended amount of Tracked Vehicles are present.</English>
```

**Review by jonpas — CHANGES_REQUESTED** [review]

**Comment by jonpas** [comment]
"> 10 units in group" check should be added [here](https://github.com/Theseus-Aegis/Mods/blob/4cbc0498a7412ec8eba344f30ca45fc6905afa62/addons/mission_autotest/functions/fnc_testGroupSizes.sqf) instead, with every group that has more than 10 units being logged so it can be found.

**Comment by jonpas** [comment]
Would also appreciate if Tank/Tracked APC/WheeledAPC can be streamlined between text and internal names.

**Review by jonpas — APPROVED** [review]

**Pull request merged**